domain/state.go

```diff
@@ -167,10 +167,10 @@ func Run(ctx AtomicContext, fn func(context.Context, *sqlair.TX) error) error {
 	// This is to prevent the transaction from being removed from the context
 	// or the service layer from attempting to use the transaction outside of
 	// the transaction scope.
-	txCtx.mu.Lock()
-	defer txCtx.mu.Unlock()
+	atomic.mu.Lock()
```

Comment:
I'd be inclined to apply the uniform access principle here and also add `lock`/'unlock` methods.